File: summary/rust/8_least_common_multiple.rs
// 最大公約数を求める関数
fn gcd(a: u64, b: u64) -> u64 {
    if b == 0 {
        a
    } else {
        gcd(b, a % b)
    }
}

// 最小公倍数を求める関数
fn lcm(a: u64, b: u64) -> u64 {
    a / gcd(a, b) * b
}

fn main() {
    let a = 15;
    let b = 20;
    println!("{} と {} の最小公倍数は {}", a, b, lcm(a, b));
}

File: summary/c++/8_least_common_multiple.cpp
#include <iostream>
#include <cmath> // For std::abs

// Function to calculate the Greatest Common Divisor (GCD)
int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

// Function to calculate the Least Common Multiple (LCM)
int lcm(int a, int b) {
    return std::abs(a * b) / gcd(a, b);
}

int main() {
    int a, b;
    std::cout << "Enter two integers: ";
    std::cin >> a >> b;
    std::cout << "The Least Common Multiple of " << a << " and " << b << " is " << lcm(a, b) << std::endl;
    return 0;
}
